src/liballoc/btree/node.rs

```diff
@@ -168,6 +184,26 @@ unsafe impl<K: Sync, V: Sync> Sync for Root<K, V> { }
 unsafe impl<K: Send, V: Send> Send for Root<K, V> { }
 
 impl<K, V> Root<K, V> {
+    pub fn is_shared_root(&self) -> bool {
+        ptr::eq(
+            unsafe { self.node.as_ptr().as_ref() },
+            &EMPTY_ROOT_NODE as *const LeafNode<(), ()> as *const LeafNode<K, V>,
+        )
```

Comment:
is there any reason to use ptr::eq and not just `==` ?

Reply:
(Also is the as_ref() necessary..?)

Reply:
`ptr::eq` just to be explicit about not deref-ing, but that's not necessary if they're raw pointers, and the `as_ref` was because I expected to compare the the reference not the raw pointer. I'll clean that up.

Reply:
Oh wait, yes, `as_ref()` is necessary because `as_ptr()` returns a `NonNull<_>`, so `ptr::eq` lets you avoid an extra pair of coercions.

I think `.as_ptr().as_ptr()` is better than `.as_ptr().as_ref()` though since then we don't need `unsafe`.